src/Core/Services/Implementations/OrganizationService.cs

```diff
@@ -1651,5 +1641,63 @@ private void ValidateOrganizationUpgradeParameters(Models.StaticStore.Plan plan,
                     $"{plan.MaxAdditionalSeats.GetValueOrDefault(0)} additional users.");
             }
         }
+
+        private async Task ValidateOrganizationUserUpdatePermissions(Guid loggedInUserId, Guid organizationId, OrganizationUserType newType, OrganizationUserType? oldType) 
+        {
+            var checkedTypes = new List<OrganizationUserType> 
+            { 
+                OrganizationUserType.Owner,
+                OrganizationUserType.Admin,
+                OrganizationUserType.Custom, 
+            };
+
+            if (checkedTypes.Contains(newType) || (oldType.HasValue && checkedTypes.Contains(oldType.Value)))
+            {
+                var loggedInUserOrgs = await _organizationUserRepository.GetManyByUserAsync(loggedInUserId);
+                var loggedInAsOrgOwner = loggedInUserOrgs
+                    .Any(u => u.OrganizationId == organizationId && u.Type == OrganizationUserType.Owner);
+                if (!loggedInAsOrgOwner)
```

Comment:
Personally, I'd invert the loggedInAs____ if statements to avoid this deep nesting